BraveWallet/Crypto/Stores/ManageSiteConnectionsStore.swift

```diff
@@ -0,0 +1,82 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import Data
+import BraveCore
+
+struct SiteConnection: Equatable, Identifiable {
+  let url: String
+  let connectedAddresses: [String]
+  
+  var id: String { url }
+}
+
+extension Array where Element == SiteConnection {
+  /// Filters the array by checking if the given `text` exists in the `url` or `connectedAccountAddresses`
+  func filter(by text: String) -> [SiteConnection] {
+    filter { siteConnections in
+      guard !text.isEmpty else { return true }
+      let filterText = text.lowercased()
```

Comment:
do we still have to lowercase it if in the line below you do a case insensitive comparison?

Reply:
We still need the lowercase here for the `siteConnections.url.contains(filterText)` check which is case-sensitive